core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -642,22 +651,38 @@ private boolean isIntermediateTopic(final String topic) {
         return options.valuesOf(intermediateTopicsOption).contains(topic);
     }
 
-    private void maybeDeleteInternalTopics(final Admin adminClient, final boolean dryRun) {
-        System.out.println("Deleting all internal/auto-created topics for application " + options.valueOf(applicationIdOption));
-        final List<String> topicsToDelete = new ArrayList<>();
-        for (final String listing : allTopics) {
-            if (isInternalTopic(listing)) {
-                if (!dryRun) {
-                    topicsToDelete.add(listing);
-                } else {
-                    System.out.println("Topic: " + listing);
-                }
+    private int maybeDeleteInternalTopics(final Admin adminClient, final boolean dryRun) {
+        final List<String> inferredInternalTopics = allTopics.stream()
+                .filter(this::isInferredInternalTopic)
+                .collect(Collectors.toList());
+        final List<String> specifiedInternalTopics = options.valuesOf(internalTopicsOption);
+        final List<String> topicsToDelete;
+
+        if (!specifiedInternalTopics.isEmpty()) {
+            final List<String> notFoundInternalTopics = specifiedInternalTopics.stream()
+                    .filter(topic -> !inferredInternalTopics.contains(topic))
+                    .collect(Collectors.toList());
+            if (!notFoundInternalTopics.isEmpty()) {
```

Comment:
something like `inferredInternalTopics.containsAll(specifiedInternalTopics)` might be easier to understand here